### PR TITLE
Add blog post on hardening the Collector

### PR DESCRIPTION
**Pull request by mx-psi**
Adds new blog post in relation to open-telemetry/opentelemetry-collector/pull/10352 and open-telemetry/opentelemetry-collector-contrib/pull/33658.

## Docs PR Checklist

- [x] This PR is for a documentation page whose authoritative copy is in the
      opentelemetry.io repository.

### CODE REVIEW

**Comment by svrnm** [comment]
@mx-psi can you create an issue for this (you can keep the PR in draft) following these updated guidelines: https://opentelemetry.io/docs/contributing/blog/#before-submitting-a-blog-post ? 

Thanks!

**Comment by mx-psi** [comment]
> @mx-psi can you create an issue for this (you can keep the PR in draft) following these updated guidelines: [opentelemetry.io/docs/contributing/blog/#before-submitting-a-blog-post](https://opentelemetry.io/docs/contributing/blog/#before-submitting-a-blog-post) ?
> 
> Thanks!

@svrnm Sure! See #4760

**Comment by mx-psi** [comment]
@svrnm if you don't mind helping here, since the publishing date is set to July 2nd I can't preview the contents. Any suggestions here? Maybe I can change the date temporarily to help people review this?

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (svrnm) [inline content/en/blog/2024/hardening-the-collector-one.md:7]:
```suggestion
cSpell:ignore: awsfirehose awsproxy awsxray Baeyens jaegerremotesampling loki remotetap sapm signalfx skywalking splunk
issue: 4760
sig: Collector SIG
```

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (theletterf) [inline content/en/blog/2024/hardening-the-collector-one.md:2]:
```suggestion
title: 'Hardening the Collector Episode 1: A new default bind address'
```

_Cue Star Wars opening fanfare_ 🥳

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (theletterf) [inline content/en/blog/2024/hardening-the-collector-one.md:15]: Perhaps a link to the changelog line or PR?

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (theletterf) [inline content/en/blog/2024/hardening-the-collector-one.md:21]: As a reader, this leaves me wondering about the baseline or current security status of the Collector. How safe is it? Can we say?

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (theletterf) [inline content/en/blog/2024/hardening-the-collector-one.md:97]:
```suggestion
want to use (for example, `status.podIP` on Kubernetes).
```

**Review by theletterf — APPROVED** [review]
Good stuff! Can you perhaps end with a note on what to expect in Episode 2?

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (mx-psi) [reply content/en/blog/2024/hardening-the-collector-one.md:21]: Would it help to link to https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md ? 

Or maybe, the security audit results would also be helpful here. I think we are going to publish them at some point, cc @jpkrohling

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (jpkrohling) [inline content/en/blog/2024/hardening-the-collector-one.md:13]: Can we use a similar wording from the CVE blog post? "sponsored by, facilitated by, executed by..."?

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (mx-psi) [reply content/en/blog/2024/hardening-the-collector-one.md]: Added a link to the changelog: 0c360aa2b127f2633a9c50de6c506acffc4612c9

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (jpkrohling) [reply content/en/blog/2024/hardening-the-collector-one.md:22]: I would add that those are recommendations, and not classified as issues or vulnerabilities. We can state that we are happy with the results, showing that secure coding practices are a solid part of our routine.

> While we expect the report to be made public soon, we can already say that we are very satisfied with the confirmation that the Collector has proven to be very secure, highlighting the secure coding practices and processes we have in place.

**Review by jpkrohling — APPROVED** [review]

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (mx-psi) [reply content/en/blog/2024/hardening-the-collector-one.md]: Done, I used the same wording

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (mx-psi) [reply content/en/blog/2024/hardening-the-collector-one.md]: > I would add that those are recommendations, and not classified as issues or vulnerabilities.

I went with 'security best practices'. I also added the sentence you mention :)

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (TylerHelmuth) [inline content/en/blog/2024/hardening-the-collector-one.md:104]: Let's add in the "what we've done" section above when the helm chart changed, and then reference here that you're not affected because "we changed the helm chart a long time ago" or something.

Comment (mx-psi) [reply content/en/blog/2024/hardening-the-collector-one.md]: Done! see ecb417d354ce4fdf7429db6e1d05c40ddf6bb314

**Review by TylerHelmuth — APPROVED** [review]

File: content/en/blog/2024/hardening-the-collector-one.md
Comment (svrnm) [inline content/en/blog/2024/hardening-the-collector-one.md:4]: Let's merge it on the provided date

**Review by svrnm — CHANGES_REQUESTED** [review]
Blocking this until July 2nd when we want to merge it

@mx-psi let me know if you want to have it published earlier (e.g. today) then I will unblock it and merge it

**Comment by mx-psi** [comment]
Thanks @svrnm! I would like to merge it once we have v0.104.0 ready, which would be Monday/Tuesday next week. I will let you know if it happens at a different date

**Comment by TylerHelmuth** [comment]
@open-telemetry/docs-maintainers the release is complete and this is ready to be merged.

**Comment by theletterf** [comment]
Merged. Thanks all!

**Pull request merged**
